driver-core/src/main/java/com/datastax/driver/core/Requests.java

```diff
@@ -569,27 +591,42 @@ public void encode(Prepare msg, ByteBuf dest, ProtocolVersion version) {
                 CBUtil.writeLongString(msg.query, dest);
 
                 if (version.compareTo(ProtocolVersion.V5) >= 0) {
-                    // Write empty flags for now, to communicate that no keyspace is being set.
-                    dest.writeInt(0);
+                    // if keyspace is present write 0x1 for prepare flags.
+                    if (msg.keyspace != null) {
+                        dest.writeInt(0x01);
+                        CBUtil.writeString(msg.keyspace, dest);
+                    } else {
+                        dest.writeInt(0x00);
+                    }
                 }
             }
 
             @Override
             public int encodedSize(Prepare msg, ProtocolVersion version) {
-                return CBUtil.sizeOfLongString(msg.query);
+                int size = CBUtil.sizeOfLongString(msg.query);
+
+                if (version.compareTo(ProtocolVersion.V5) >= 0 && msg.flags.contains(QueryFlag.KEYSPACE))
+                    size += CBUtil.sizeOfString(msg.keyspace);
+                return size;
             }
         };
 
+        private final EnumSet<QueryFlag> flags = EnumSet.noneOf(QueryFlag.class);
```

Comment:
Will get rid of this since it's unused.   Originally i used it, but later noticed that QueryFlag.KEYSPACE (0x80) != 0x1 which is what prepare uses for keyspace